python/ray/air/config.py

```diff
@@ -248,47 +247,38 @@ def as_placement_group_factory(self) -> "PlacementGroupFactory":
         trainer_bundle = self._trainer_resources_not_none
         worker_bundle = self._resources_per_worker_not_none
 
-        if self.num_workers:
-            # Colocate Trainer and rank0 worker by merging their bundles
-            combined_bundle = dict(Counter(trainer_bundle) + Counter(worker_bundle))
-            bundles = [{}, combined_bundle] + [worker_bundle] * (self.num_workers - 1)
-        else:
-            bundles = [trainer_bundle]
+        # Colocate Trainer and rank0 worker by merging their bundles
+        combined_bundle = dict(Counter(trainer_bundle) + Counter(worker_bundle))
+        bundles = [{}, combined_bundle] + [worker_bundle] * (self.num_workers - 1)
```

Comment:
**Note for the future:** This empty bundle is required so that the Tune actor manager schedules the Trainable onto the combined bundle _while taking none of its resources_ rather than a non-empty head bundle.

https://github.com/ray-project/ray/blob/783da640a20ddbd3b41b893485abf187f0f27223/python/ray/air/execution/resources/request.py#L236-L244

Reply:
Added as a comment.